tests/test_basic.py

```diff
@@ -381,7 +381,7 @@ def test_ref_eq(self):
 
     def test_ref_hash(self):
         s = hash("AWS::NoValue")
-        r = hash(Ref(s))
+        r = hash(Ref("AWS::NoValue"))
```

Comment:
This worked on Python 2, but failed on Python 3.

Python 2.7.18:

```python
>>> hash("AWS::NoValue")
-6577466268859890267
>>> hash(hash("AWS::NoValue"))
-6577466268859890267
```

Python 3.8.6

```python
>>> hash("AWS::NoValue")
-3204540641682287495
>>> hash(hash("AWS::NoValue"))
-898697632468593544
>>> hash(hash(hash("AWS::NoValue")))
-898697632468593544
```

ref cloudtools/troposphere#1053